File: guides/micronaut-openapi-swagger-ui/java/src/main/java/example/micronaut/Option.java
package example.micronaut;

import io.micronaut.serde.annotation.Serdeable;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@Serdeable // <1>
public record Option(@NotNull Language language, // <2>
                     @NotNull BuildTool buildTool, // <2>
                     @NotBlank String url) { // <2>
}

File: guides/micronaut-openapi-base/python/src/example/micronaut/option.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank, NotNull
from micronaut.serde.annotation import Serdeable

from .build_tool import BuildTool
from .language import Language


@Serdeable  # <1>
@dataclass
class Option:
    language: Annotated[Language, NotNull]  # <2>
    buildTool: Annotated[BuildTool, NotNull]  # <2>
    url: Annotated[str, NotBlank]  # <2>
